ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -469,7 +470,7 @@ static KsqlRestApplication buildApplication(
         versionCheckerFactory,
         Integer.MAX_VALUE,
         serviceContext,
-        KsqlSecurityContextBinder::new);
+        KsqlSecurityContextBinder.createFactory(serviceContext));
```

Comment:
I think this `createFactory` method is not necessary. The default `serviceContext` is passed to the `KsqlRestApplication` in the previous line. With that, you can configure the `KsqlSecurityContextBinder` in the `configureBaseApplication` method like:

```
config.register(serviceContextBinderFactory.apply(ksqlConfigNoPort, securityExtension, serviceContext));
```

Reply:
Ah, yeah.  I was kind of trying to fit the existing types, but you're right.  Simplified to:
```
config.register(new KsqlSecurityContextBinder(
        ksqlConfigNoPort, securityExtension, serviceContext));
```

Reply:
Doesn't work quite that way, per our below discussion.